apps/examples/rtc_alarm/rtc_alarm_main.c

```diff
@@ -0,0 +1,353 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/ioctl.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sched.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include <tinyara/rtc.h>
+
+#ifdef CONFIG_RTC_ALARM
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static bool g_alarm_daemon_started;
+static pid_t g_alarm_daemon_pid;
+static bool g_alarm_received[CONFIG_RTC_NALARMS];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: alarm_handler
+ ****************************************************************************/
+
+static void alarm_handler(int signo, FAR siginfo_t *info, FAR void *ucontext)
+{
+	int almndx = info->si_value.sival_int;
+	if (almndx >= 0 && almndx < CONFIG_RTC_NALARMS) {
+		g_alarm_received[almndx] = true;
+	}
+}
+
+/****************************************************************************
+ * Name: alarm_daemon
+ ****************************************************************************/
+
+static int alarm_daemon(int argc, FAR char *argv[])
```

Comment:
We dont need separate daemon. Just register the signal handler for the main app only. 
We dont need to wait for alarn in loop. It defeats the purpose of having alarm. 
Just register the signal and when callback is called, print the info that alarm has expired.